src/inmanta/data/__init__.py

```diff
@@ -1571,6 +1571,26 @@ async def insert(self, connection: Optional[asyncpg.connection.Connection] = Non
         )
         await self._execute_query(query, *values, connection=connection)
 
+    async def insert_with_overwrite(self, connection: Optional[asyncpg.connection.Connection] = None) -> None:
```

Comment:
There is a lot of code duplication between the `insert_with_overwrite` method and the `insert_many_with_overwrite` method. The latter is just an extended version of the former. If we let the `insert_with_overwrite` method delegate to `insert_many_with_overwrite`, by passing in just one document, we can get rid of a lot of code.